<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Users</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

Linux system mein user wo banda ya process hota hai jo system ko use karta hai 
 - file read/write karta hai
 - apps chalata hai, ya
 - commands execute karta hai.

Har user ka lye
- Username
- User ID (UID)
- Home directory
- Shell
- Permissions (read/write/execute)
  
Hoti hn

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Types of Users</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 1 Root User (Superuser)
- Sab kuch access karta hai (full power)
- UID = 0
  
Note ! Normal Work ky liye Use mat karo

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

#### Login Root User

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

 `su -`
- Matlab: Switch to root user, with root’s full environment.
- Security: Agar kisi ne root ka password le liya to full control mil jata hai.
- Use Case: Server ya CLI environment mein full root shell chahiye ho.

<img src="./images/53.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`sudo su`
- Matlab: Apna sudo user ka password dekar root shell access kr sakhty hn (safe because sudoers check hota hai).
- Seurity: Better than su, kyunki sirf authorized sudo users hi use kar sakte hain.
- Use Case: Daily use ke liye best, jab full shell chahiye ho.

<img src="./images/54.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

Note! Ek Dafa login kr lia to phir Next time login karny ky lye password Required nahi ho ga

`exit`
- use `exit` command for exit cuurent user

<img src="./images/55.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`sudo -k`
- sudo ka cached password/session expire karta hai
- yani agar tumne pehle sudo use kiya aur password diya tha, to uska temporary session invalidate kar deta hai.

<img src="./images/56.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`login on startup ttf`

- #### Linux mein multiple TTYs hote hain
- #### har ek ek independent terminal session hota hai.
    
| Shortcut        | Terminal | Description                        |
| --------------- | -------- | ---------------------------------- |
| Ctrl + Alt + F1 | TTY1     | GUI login (ya CLI login)           |
| Ctrl + Alt + F2 | TTY2     | CLI Login screen (text login)      |
| Ctrl + Alt + F3 | TTY3     | CLI Login screen (text login)      |
| ...             | ...      | ...                                |
| Ctrl + Alt + F7 | TTY7     | GUI session hota tha (old distros) |


--- 
### Example

- `login: username`

- `password: password`


<img src="./images/57.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 2 Normal Users (Regular Users)
- Limited access hota hai
- Apni home directory mein kaam kar sakta hai

<img src="./images/58.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`whoami`
-  Yani, current terminal session mein kaunsa user logged in hai, uska username dikhata hai.
-  Ye command `id -un` ka shortcut hai.

In [19]:
whoami

mubeen


In [21]:
id -un

mubeen


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 3 System Users / Service Users

- Background processes ke liye
- Inka koi human interaction nahi hota
- Ye /etc/passwd mein hote hain par login nahi karte

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### Example

| User Name          | Service/Process           | Work                   |
| ------------------ | ------------------------- | -------------------------------------- |
| `mysql`            | MySQL Database            | Databases handle karta hai             |
| `www-data`         | Web server (Apache/Nginx) | Website files serve karta hai          |
| `systemd-timesync` | Time Sync daemon          | System time internet se sync karta hai |
| `nobody`           | Minimal permission user   | Bohot restricted tasks karta hai       |


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`/etc/passwd`
- Check all users:

In [2]:
cat /etc/passwd

root:x:0:0::/root:/usr/bin/bash
bin:x:1:1::/:/usr/bin/nologin
daemon:x:2:2::/:/usr/bin/nologin
mail:x:8:12::/var/spool/mail:/usr/bin/nologin
ftp:x:14:11::/srv/ftp:/usr/bin/nologin
http:x:33:33::/srv/http:/usr/bin/nologin
nobody:x:65534:65534:Kernel Overflow User:/:/usr/bin/nologin
dbus:x:81:81:System Message Bus:/:/usr/bin/nologin
systemd-coredump:x:980:980:systemd Core Dumper:/:/usr/bin/nologin
systemd-network:x:979:979:systemd Network Management:/:/usr/bin/nologin
systemd-oom:x:978:978:systemd Userspace OOM Killer:/:/usr/bin/nologin
systemd-journal-remote:x:977:977:systemd Journal Remote:/:/usr/bin/nologin
systemd-resolve:x:976:976:systemd Resolver:/:/usr/bin/nologin
systemd-timesync:x:975:975:systemd Time Synchronization:/:/usr/bin/nologin
tss:x:974:974:tss user for tpm2:/:/usr/bin/nologin
uuidd:x:973:973:UUID generator helper daemon:/var/lib/libuuid:/usr/bin/nologin
alpm:x:972:972:Arch Linux Package Management:/:/usr/bin/nologin
mubeen:x:1000:1000::/home/mubeen:/usr/bin/bash
git:x:

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

| User                     | UID   | Login Shell          | Type       | Work                           |
| ------------------------ | ----- | -------------------- | ---------- | ---------------------------------------------- |
| `root`                   | 0     | `/usr/bin/bash`      | Superuser  | Full control of system (Admin)                 |
| `bin`                    | 1     | `/usr/bin/nologin`   | System     | Binaries ka owner (historic)                   |
| `daemon`                 | 2     | `/usr/bin/nologin`   | System     | Background processes ke liye                   |
| `mail`                   | 8     | `/usr/bin/nologin`   | System     | Email spool management (old systems)           |
| `ftp`                    | 14    | `/usr/bin/nologin`   | System     | FTP server ke liye                             |
| `http`                   | 33    | `/usr/bin/nologin`   | System     | Web server (Apache/Nginx)                      |
| `nobody`                 | 65534 | `/usr/bin/nologin`   | System     | Least privilege user (temporary task)          |
| `dbus`                   | 81    | `/usr/bin/nologin`   | System     | DBus IPC system ke liye                        |
| `systemd-coredump`       | 980   | `/usr/bin/nologin`   | System     | Crashed programs ka core dump handle karta hai |
| `systemd-network`        | 979   | `/usr/bin/nologin`   | System     | Network services manage karta hai              |
| `systemd-oom`            | 978   | `/usr/bin/nologin`   | System     | OOM killer – memory control                    |
| `systemd-journal-remote` | 977   | `/usr/bin/nologin`   | System     | Logs remote machines se collect karta hai      |
| `systemd-resolve`        | 976   | `/usr/bin/nologin`   | System     | DNS resolving (systemd-resolved)               |
| `systemd-timesync`       | 975   | `/usr/bin/nologin`   | System     | Time sync service                              |
| `tss`                    | 974   | `/usr/bin/nologin`   | System     | TPM (Trusted Platform Module) ke liye          |
| `uuidd`                  | 973   | `/usr/bin/nologin`   | System     | UUID generation service                        |
| `alpm`                   | 972   | `/usr/bin/nologin`   | System     | Pacman/ALPM backend (Arch Linux)               |
| `mubeen`                 | 1000  | `/usr/bin/bash`      | **Normal** | Tumhara main user                              |
| `git`                    | 971   | `/usr/bin/git-shell` | System     | Git daemon ke liye (no normal shell)           |
| `ali`                    | 1001  | `/usr/bin/bash`      | **Normal** | Another user    |


<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Users identity and access rights </h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 1 Username

- Ye wo naam hota hai jo user ko identify karta hai.
- Har user ka ek unique username hota hai.

---

#### Example :

```bash
mubeen
ali
root
```

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        
`whoami`
- Command to check username

In [12]:
whoami

mubeen


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 2 UID (User ID) 

- System username nahi, balkay UID use karta hai users ko internally pehchanne ke liye.
---

#### Example :

| Username | UID  | Type        |
| -------- | ---- | ----------- |
| root     | 0    | Superuser   |
| mubeen   | 1000 | Normal User |
| http     | 33   | System User |


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        
`id -u`
- Command to check uid

In [16]:
id -u

1000


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 3 Home Directory – (User ka personal folder)

- Ye wo folder hota hai jahan user ka data hota hai — jaise documents, config files, etc.
    
---

| User   | Home Directory |
| ------ | -------------- |
| root   | `/root`        |
| mubeen | `/home/mubeen` |
| ali    | `/home/ali`    |


In [21]:
ls /

bin   dev  home  lib64	     media  opt   root	sbin  sys  usr
boot  etc  lib	 lost+found  mnt    proc  run	srv   tmp  var


In [19]:
ls /home

ali  lost+found  mubeen


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 4 SHELL

- Shell ka matlab hota hai ky Kaunsa command processor user use karta hai
- Zyada tar systems mein yeh hota hai:

```bash
    /bin/bash
    /usr/bin/zsh
    /usr/bin/fish
```

#### Agar koi user ka shell /usr/bin/nologin ho to:
- #### Wo login nahi kar sakta (system ya service user hota hai)

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">
        
`echo $SHELL`

- Command to check current shell

In [22]:
echo $SHELL

/usr/bin/bash


<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 5 Permissions 

- Linux mein har file/folder ke 3 type ke permission holders hote hain:

| Type   | Means               |
| ------ | -------------------- |
| Owner  | File ka owner (user) |
| Group  | File se linked group |
| Others | Baaki sab users        |

- Aur 3 type ki permissions hoti hain:

| Symbol | Meaning |
| ------ | ------- |
| r      | Read    |
| w      | Write   |
| x      | Execute |


In [25]:
ls -l

total 52
-rw-r--r-- 1 mubeen mubeen    0 Jun  7 20:52  1.txt
drwxr-xr-x 2 mubeen mubeen 4096 Jun  7 20:53  Documents
drwxr-xr-x 2 mubeen mubeen 4096 Jun  7 20:52 'My Files'
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 11:05 'This is my folder'
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 11:03  a
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 11:03  b
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 11:03  c
drwxr-xr-x 3 mubeen mubeen 4096 Jun  8 12:05  code
drwxr-xr-x 3 mubeen mubeen 4096 Jun  8 10:55  folder1
drwx---rwx 6 mubeen mubeen 4096 Jun 22 18:18  linux
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 09:16  myfolder
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 10:59  new_folder
drwxr-xr-x 4 mubeen mubeen 4096 Jun 14 15:08  watchlist
drwxr-xr-x 2 mubeen mubeen 4096 Jun  8 11:10  xyz


<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Create User, Set Password and Remove User</h1>

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 1 Create a Normal User

`sudo useradd -m user1`

- `-m` parameter User ka home folder bhi create kary ga

<img src="./images/59.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<body style="font-family: Sans-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 2 set User Password

`sudo passwd user1`

- password pucha jaayega, 2 dafa likhna hoga.

<img src="./images/60.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<body style="font-family: Sansa-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 3. Shell Check (Login Enable Hai Ya Nahi)

`grep user1 /etc/passwd`


- Agar /usr/bin/nologin hai to login disable hoga

In [33]:
!grep user1 /etc/passwd

user1:x:1002:1002::/home/user1:/usr/bin/bash


<body style="font-family: Sansa-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 4 login user

`su user1`

<img src="./images/61.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<body style="font-family: Sansa-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

### 5 User remove 

`sudo del user1`

- Ye command user user1 ko remove kar dega, lekin uska home folder (/home/user1) system mein reh jaayega.

<body style="font-family: Sansa-serif;">
    <div style="color: black; font-size: 15px; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px black; padding: 20px;">

`sudo del -r user1`

- `-r` `parameter` User ka home directory bhi delete karega (`/home/user1`)
- ⚠️ Ye user ka sara personal data delete kar dega — careful rahna!


<img src="./images/62.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Users Mindmap</h1>

<img src="./images/63.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Root User Mindmap</h1>

<img src="./images/64.jpg" style="display: block;margin-left: auto; margin-right: auto;width:70%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
Normal User and Syestem, Service User Mindmap</h1>

<img src="./images/65.jpg" style="display: block;margin-left: auto; margin-right: auto;width:100%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
User Identity and Access Rights Mindmap</h1>

<img src="./images/66.jpg" style="display: block;margin-left: auto; margin-right: auto;width:100%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
User Management Operations Mindmap</h1>

<img src="./images/67.jpg" style="display: block;margin-left: auto; margin-right: auto;width:100%">

<h1 style="text-align:center; color:#005bbd; font-size:20px; font-family:Sans-serif; font-style: oblique; text-shadow: 0 0 3px white, 0 0 1px Black;">
/etc/password and Commands</h1>

<img src="./images/68.jpg" style="display: block;margin-left: auto; margin-right: auto;width:100%">